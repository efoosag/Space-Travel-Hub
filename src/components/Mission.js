/* eslint-disable object-curly-newline */
/* eslint-disable comma-dangle */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions';
import '../App.css';

const Mission = ({ mission }) => {
  const { id, name, description, joined } = mission;

  const dispatch = useDispatch();

  return (
    <>
      <td className="centerAlign">{name}</td>
      <td className="description">{description}</td>
      <td className="centerAlign">
        <span>
          {joined ? (
            <span className="badge-leave">NOT A MEMBER</span>
          ) : (
            <span className="badge-join">ACTIVE MEMBER</span>
          )}
        </span>
      </td>
      <td className="centerAlign">
        {joined ? (
          <button
            type="submit"
            className="btn-join"
            onClick={() => dispatch(joinMission({ id }))}
          >
            JOIN MISSION
          </button>
        ) : (
          <button
            type="submit"
            className="btn-leave"
            onClick={() => dispatch(leaveMission({ id }))}
          >
            LEAVE MISSION
          </button>
        )}
      </td>
    </>
  );
};

Mission.propTypes = {
  mission: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string,
    joined: PropTypes.bool,
  }),
};

Mission.defaultProps = {
  mission: null,
};

export default Mission;
