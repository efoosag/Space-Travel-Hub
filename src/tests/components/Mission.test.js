import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Mission from '../../components/Mission';

describe('Testing Mission component', () => {
  it('Testing with snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Mission
          mission={{
            mission_id: 2,
            mission_name: 'Telstar',
            description:
              'Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.',
            reserved: false,
          }}
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
