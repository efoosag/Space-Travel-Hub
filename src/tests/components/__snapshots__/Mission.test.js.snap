// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission component Testing with snapshot 1`] = `
<tr>
  <td
    className="centerAlign"
  >
    Falcon 1
  </td>
  <td
    className="description"
  >
    The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
  </td>
  <td
    className="centerAlign"
  >
    <span>
      <span
        className="badge-leave"
      >
        ACTIVE MEMBER
      </span>
    </span>
  </td>
  <td
    className="centerAlign"
  >
    <button
      className="btn-leave"
      onClick={[Function]}
      type="submit"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
